js/reference.js

```diff
@@ -621,12 +622,113 @@ var ERMrest = (function(module) {
          * Updates a set of resources.
          * @param {!Array} tbd TBD parameters. Probably an array of pairs of
          * [ (keys+values, allvalues)]+ ] for all entities to be updated.
-         * @returns {Promise} A promise for a TBD result or errors.
+         * @returns {Promise} page A promise for a page result or errors.
          */
-        update: function(tbd) {
+        update: function(tuples) {
             try {
-                // TODO
-                notimplemented();
+                verify(tuples, "'tuples' must be specified");
+                verify(tuples.length > 0, "'tuples' must have at least one row to create");
+
+                var defer = module._q.defer();
+
+                var self = this,
+                    uri = this._location.service + "/catalog/" + this._location.catalog + "/attributegroup/" + this._location.schemaName + ':' + this._location.tableName + '/';
+
+                var submissionData = [],
+                    columnProjections = [],
+                    shortestKeyNames = [],
+                    keyWasModified = false,
+                    tuple, oldData, newData, keyName;
+
+
+                shortestKeyNames = this._shortestKey.map(function (column) {
+                    return column.name;
+                });
+
+                tuples.forEach(function (tuple) {
+                    newData = tuple.data;
+                    oldData = tuple._oldData;
+                    for (var key in newData) {
+                        // if the key is part of the shortest key for the entity, the data needs to be aliased
```

Comment:
Don't we need to alias all columns so that we avoid possible name collisions? Say an the table definition includes columns named: `filename` and `old_filename` or `new_filename` and they use it as a key.

I would like to see your test schema include key and non-key columns with the `old_` and `new_` prefixes in them.